src/picongpu/include/particles/bremsstrahlung/Bremsstrahlung.tpp

```diff
@@ -0,0 +1,252 @@
+/**
+ * Copyright 2016 Heiko Burau
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+namespace picongpu
+{
+namespace particles
+{
+namespace bremsstrahlung
+{
+
+template<typename T_IonSpecies, typename T_ElectronSpecies, typename T_PhotonSpecies>
+Bremsstrahlung<T_IonSpecies, T_ElectronSpecies, T_PhotonSpecies>::Bremsstrahlung(
+    const ScaledSpectrum::LookupTableFunctor& scaledSpectrumFunctor,
+    const ScaledSpectrum::LookupTableFunctor& stoppingPowerFunctor,
+    const GetPhotonAngle::GetPhotonAngleFunctor& getPhotonAngleFunctor,
+    const uint32_t currentStep)
+        : scaledSpectrumFunctor(scaledSpectrumFunctor),
+          stoppingPowerFunctor(stoppingPowerFunctor),
+          getPhotonAngleFunctor(getPhotonAngleFunctor),
+          photonMom(float3_X::create(0)),
+          randomGen(RNGFactory::createRandom<Distribution>())
+{
+    DataConnector &dc = Environment<>::get().DataConnector();
+
+    /* initialize pointers on host-side tmp-field databoxes */
+    FieldTmp* fieldIonCounter = &(dc.getData<FieldTmp > (FieldTmp::getName(), true));
+    /* reset values to zero */
+    fieldIonCounter->getGridBuffer().getDeviceBuffer().setValue(FieldTmp::ValueType(0.0));
+
+    /* load species without copying the particle data to the host */
+    T_IonSpecies* ionSpecies = &(dc.getData<T_IonSpecies >(T_IonSpecies::FrameType::getName(), true));
+
+    /* compute ion density */
+    typedef typename CreateCounterOperation<T_IonSpecies>::type::Solver CounterSolver;
+    fieldIonCounter->computeValue < CORE + BORDER, CounterSolver > (*ionSpecies, currentStep);
+    dc.releaseData(T_IonSpecies::FrameType::getName());
+
+    /* initialize device-side tmp-field databoxes */
+    this->ionCounterBox = fieldIonCounter->getDeviceDataBox();
+
+    const float_64 softPhotonsCutoff_SI = photon::SOFT_PHOTONS_CUTOFF_keV * UNITCONV_keV_to_Joule;
+    this->softPhotonsCutoff = softPhotonsCutoff_SI / UNIT_ENERGY;
```

Comment:
This is in SI units! Please move these two lines in `param`/`unitless` and use the unitless value here only.